theano/tensor/signal/pool.py

```diff
@@ -234,50 +232,86 @@ def out_shape(imgshape, ds, ignore_border=False, st=None, padding=(0, 0)):
         rval = list(imgshape[:-2]) + [nr, nc]
         return rval
 
-    def __init__(self, ds, ignore_border=False, st=None, padding=(0, 0),
-                 mode='max', openmp=None):
+    def __init__(self, ignore_border=False, mode='max', openmp=None):
         super(Pool, self).__init__(openmp=openmp)
-        self.ds = tuple(ds)
-        if not all([isinstance(d, integer_types) for d in ds]):
-            raise ValueError(
-                "Pool downsample parameters must be ints."
-                " Got %s" % str(ds))
-        if st is None:
-            st = ds
-        assert isinstance(st, (tuple, list))
-        self.st = tuple(st)
         self.ignore_border = ignore_border
-        self.padding = tuple(padding)
-        if self.padding != (0, 0) and not ignore_border:
-            raise NotImplementedError(
-                'padding works only with ignore_border=True')
-        if self.padding[0] >= self.ds[0] or self.padding[1] >= self.ds[1]:
-            raise NotImplementedError(
-                'padding_h and padding_w must be smaller than strides')
         if mode not in ['max', 'average_inc_pad', 'average_exc_pad', 'sum']:
             raise ValueError(
                 "Pool mode parameter only support 'max', 'sum',"
                 " 'average_inc_pad' and 'average_exc_pad'. Got %s" % mode)
         self.mode = mode
 
-    def make_node(self, x):
+    def prepare_node(self, node, storage_map, compute_map):
+        if len(node.inputs) == 1:
+            warnings.warn("Theano Pool internal changed.", stacklevel=3)
```

Comment:
Add something like "We convert it" at the end so that people don't panic with this. Or just remove it as we do it automatically? I would just remove it, but if you prefer to keep it, update the message to help prevent user questions.